src/crypto.js

```diff
@@ -0,0 +1,48 @@
+const libsodium = require('libsodium-wrappers');
+
+export class Crypto {
+  constructor() {
+    return (async () => {
+      await libsodium.ready;
+      this._sodium = libsodium;
+
+      return this;
+    })();
+  }
+
+  /*
+  *  Receives buffer attributes. Returns buffer ciphertext.
+  *  data attr format can be of any type (string, buffer)
+  */
+  encrypt(data, pk, nonce) {
```

Comment:
Minor but lets align these method signatures with those of PorkyLib:

```ruby
encrypt_with_key(plaintext, plaintext_key)
```

The nonce can be generated by this function. The PorkyLib function returns both the ciphertext and nonce.